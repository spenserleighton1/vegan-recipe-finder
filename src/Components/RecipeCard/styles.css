.recipe-card {
  margin: auto;
  font-family: 'Cinzel', serif;
  width: 20vw;
  padding: 10px;
}

.recipe-img {
  height: 200px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.recipe-btn {
  color: #3F493F;
  font-size: 18px;
  font-family: 'Cinzel', serif;
  background-color: transparent;
  border: none;
}

.recipe-btn:hover {
  color: darkgreen;
}