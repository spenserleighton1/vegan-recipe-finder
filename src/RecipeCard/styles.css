.recipe-card {
  width: 30vw;
  height: 500px
}

.recipe-img {
  height: 300px;
}

.recipe-btn {
  display: block;
}