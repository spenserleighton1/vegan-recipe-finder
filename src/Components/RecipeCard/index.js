import React from 'react';
import './styles.css';
import { apiKey } from '../../helper/apiKey';
import PropTypes from 'prop-types';

const RecipeCard = ({ recipeTitle, recipeImage, fetchRecipe, recipeId }) => {
  return (
    <div className='recipe-card'>
      <h3 className='recipe-title'>{ recipeTitle }</h3>
      <img className='recipe-img' src={ recipeImage } alt={ recipeTitle } />
      <button className='recipe-btn' onClick={ () => { fetchRecipe(apiKey, recipeId); } }>view recipe</button>
    </div>
  );
};

export default RecipeCard;
      
RecipeCard.propTypes = {
  recipeTitle: PropTypes.string,
  recipeImage: PropTypes.string,
  fetchRecipe: PropTypes.func,
  recipeId: PropTypes.number
};