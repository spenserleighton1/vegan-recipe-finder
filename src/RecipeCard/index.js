import React from 'react';
import './styles.css'
import { apiKey } from '../helper/apiKey';

const RecipeCard = ({ recipeTitle, recipeLink, recipeImage, fetchRecipe, recipeId }) => {
  return (
    <div className='recipe-card'>
      <h3 className='recipe-title'>{ recipeTitle }</h3>
      <img className='recipe-img' src={ recipeImage } alt={ recipeTitle } />
      <button className='recipe-btn' onClick={ () => { fetchRecipe(apiKey, recipeId) } }>view recipe</button>
    </div>
    )
}

export default RecipeCard;
      
      