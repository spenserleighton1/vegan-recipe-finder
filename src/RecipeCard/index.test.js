import React from 'react';
import RecipeCard from './'
import { shallow } from 'enzyme';

describe('RecipeCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<RecipeCard />)

    expect(wrapper).toMatchSnapshot();
  })
})