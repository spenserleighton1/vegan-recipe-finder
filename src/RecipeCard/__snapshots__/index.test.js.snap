// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="recipe-title"
/>,
        <img
          alt={undefined}
          className="recipe-img"
          src={undefined}
/>,
        <a
          className="recipe-link"
          href={undefined}
>
          View Recipe
</a>,
      ],
      "className": "recipe-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "recipe-title",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "recipe-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View Recipe",
          "className": "recipe-link",
          "href": undefined,
        },
        "ref": null,
        "rendered": "View Recipe",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="recipe-title"
/>,
          <img
            alt={undefined}
            className="recipe-img"
            src={undefined}
/>,
          <a
            className="recipe-link"
            href={undefined}
>
            View Recipe
</a>,
        ],
        "className": "recipe-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "recipe-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "recipe-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View Recipe",
            "className": "recipe-link",
            "href": undefined,
          },
          "ref": null,
          "rendered": "View Recipe",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
